app/main.py

```diff
@@ -1,34 +1,53 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.decks = []
+        for row_index in range(start[0], end[0] + 1):
+            for column_index in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row_index, column_index))
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self,
+                 row: int,
+                 column: int) -> Deck | None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self,
+             row: int,
+             column: int) -> None:
+        self.get_deck(row, column).is_alive = False
```

Comment:
self.get_deck(row, column)
can return None
First of all you must check that desk is not None

Reply:
@HalynaPetrova not fixed

Reply:
You must check here if self.get_deck() return None